x/ecocredit/core/msg_bridge.go

```diff
@@ -0,0 +1,49 @@
+package core
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/x/auth/legacy/legacytx"
+	"github.com/regen-network/regen-ledger/x/ecocredit"
+)
+
+var _ legacytx.LegacyMsg = &MsgBridge{}
+
+// Route implements the LegacyMsg interface.
+func (m MsgBridge) Route() string { return sdk.MsgTypeURL(&m) }
+
+// Type implements the LegacyMsg interface.
+func (m MsgBridge) Type() string { return sdk.MsgTypeURL(&m) }
+
+// GetSignBytes implements the LegacyMsg interface.
+func (m MsgBridge) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ecocredit.ModuleCdc.MustMarshalJSON(&m))
+}
+
+// ValidateBasic does a sanity check on the provided data.
+func (m *MsgBridge) ValidateBasic() error {
+
+	if err := m.MsgCancel.ValidateBasic(); err != nil {
+		return err
+	}
+
+	if len(m.BridgeContract) == 0 {
+		return sdkerrors.ErrInvalidRequest.Wrap("bridge contract should not be empty")
+	}
+
+	if len(m.BridgeRecipient) == 0 {
+		return sdkerrors.ErrInvalidRequest.Wrap("bridge recipient address should not be empty")
+	}
+
+	if len(m.BridgeTarget) == 0 {
+		return sdkerrors.ErrInvalidRequest.Wrap("bridge target should not be empty")
+	}
```

Comment:
We might want to consider defining supported targets and validate against those. We could have enumerated values stored in the proto files (similar to how we do digest algorithm, media type, etc. in the date module). We would only support Polygon in this initial version. We should be as cautious as we can given cancelling credits cannot be undone. Ok to leave without for now but something to discuss.

Reply:
maybe not a bad idea to prevent unexpected fund transfers. should be gov gated in the future IMHO so a chain upgrade isn't always needed but maybe okay for now to just allow `polygon-tco2`

Reply:
if we're doing validation here, I prefer `polygon` as opposed to `polygon-tco2`, as we don't want a situation where there could be another way of interpreting the same polygon txhash for creating a different kind of ecocredit / bridge operation.